src/main/java/org/mastodon/views/trackscheme/display/PaintGraph.java

```diff
@@ -497,37 +512,23 @@ else if ( specifiedColor == 0 )
 					? 1 - completionRatio
 					: completionRatio;
 			final boolean fade = ( transition == APPEAR || transition == DISAPPEAR );
-			int r, g, b, a;
-			if ( specifiedColor == 0 )
-			{
-				r = normalColor.getRed();
-				g = normalColor.getGreen();
-				b = normalColor.getBlue();
-				a = normalColor.getAlpha();
-			}
-			else
-			{
-				r = ARGBType.red( specifiedColor );
-				g = ARGBType.green( specifiedColor );
-				b = ARGBType.blue( specifiedColor );
-				a = ARGBType.alpha( specifiedColor );
-			}
+			int rgb = specifiedColor == 0 ? normalColor.getRGB() : specifiedColor;
 			if ( isSelected || !fade )
-			{
-				r = ( int ) ( ratio * r + ( 1 - ratio ) * selectedColor.getRed() );
-				g = ( int ) ( ratio * g + ( 1 - ratio ) * selectedColor.getGreen() );
-				b = ( int ) ( ratio * b + ( 1 - ratio ) * selectedColor.getBlue() );
-				a = ( int ) ( ratio * a + ( 1 - ratio ) * selectedColor.getAlpha() );
-			}
+				rgb = ColorUtils.mixColors( selectedColor.getRGB(), rgb, ( float ) ratio );
 			if ( fade )
-				a = ( int ) ( a * ( 1 - ratio ) );
-			final Color color = new Color( r, g, b, a );
+				rgb = ColorUtils.scaleAlpha( rgb, (float) (1 - ratio) );
+			final Color color = new Color( rgb );
 			return isGhost
 					? TrackSchemeStyle.mixGhostColor( color, style.getBackgroundColor() )
 					: color;
 		}
 	}
 
+	protected boolean isFadingFutureTimepoints()
+	{
+		return false;
+	}
```

Comment:
This method here is part of a hack. You inject the code that is actually needed for the PaintHierarchyGraph into this class but you dissable it by using the `isFadingGutureTimepoints()` method. Now there is a simple solution for this ScreenVertex.isFaded should always be false.

Reply:
I removed the method.